Client/Frontend/Settings/HomePageSettingsViewController.swift

```diff
@@ -1,9 +1,126 @@
+        tableView.registerClass(SettingsTableSectionHeaderFooterView.self, forHeaderFooterViewReuseIdentifier: SectionHeaderFooterIdentifier)
+
+        tableView.separatorColor = UIConstants.TableViewSeparatorColor
+        tableView.backgroundColor = UIConstants.TableViewHeaderBackgroundColor
+        let footer = SettingsTableSectionHeaderFooterView(frame: CGRect(x: 0, y: 0, width: tableView.bounds.width, height: UIConstants.TableViewHeaderFooterHeight))
+        footer.showBottomBorder = false
+        tableView.tableFooterView = footer
+    }
+
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        if indexPath.section == SectionSettings {
+            let setting = settings[indexPath.row]
+            var cell: UITableViewCell!
+            if let _ = setting.status {
+                // Work around http://stackoverflow.com/a/9999821 and http://stackoverflow.com/a/25901083 by using a new cell.
+                // I could not make any setNeedsLayout solution work in the case where we disconnect and then connect a new account.
+                // Be aware that dequeing and then ignoring a cell appears to cause issues; only deque a cell if you're going to return it.
+                cell = UITableViewCell(style: setting.style, reuseIdentifier: nil)
+            } else {
+                cell = tableView.dequeueReusableCellWithIdentifier(SectionSettingsIdentifier, forIndexPath: indexPath)
+            }
+            setting.onConfigureCell(cell)
+            return cell
+        } else {
+            let cell = UITableViewCell(style: UITableViewCellStyle.Default, reuseIdentifier: nil)
+            assert(indexPath.section == SectionURLTextView)
+            cell.textLabel?.text = Strings.SettingsHomePageURLSectionTitle
+            cell.textLabel?.textAlignment = NSTextAlignment.Natural
+            cell.textLabel?.textColor = UIConstants.TableViewRowTextColor
+            cell.accessibilityTraits = UIAccessibilityTraitButton
+            cell.textLabel?.text = UIConstants.DefaultHomePage.absoluteDisplayString()
+            cell.accessibilityIdentifier = "HomePage"
+            clearButton = cell
+            return cell
+        }
+
+    }
+
+    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return NumberOfSections
+    }
+
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if section == SectionSettings {
+            return settings.count
+        }
+
+        assert(section == SectionURLTextView)
+        return 1
+    }
+
+    override func tableView(tableView: UITableView, shouldHighlightRowAtIndexPath indexPath: NSIndexPath) -> Bool {
+        guard indexPath.section == SectionURLTextView else { return false }
+
+        // Highlight the button only if it's enabled.
+        return false
```

Comment:
if we're returning false regardless of whether or not the `guard` succeeds, why do we need the `guard`?